kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/AbstractDataset.java

```diff
@@ -150,6 +150,12 @@ public boolean deleteAll() {
   }
 
   @Override
+  public boolean canDelete() {
+    throw new UnsupportedOperationException(
+            "This Dataset does not support checking deletion");
```

Comment:
Should this just return false rather than throwing and exception? I don't see user code making a distinction between "can't delete this" and "delete operation not supported for this", but there may be a flow I'm missing.